UX Editor button to simplify the insertion of a module on an article

dgrammatiko opened the pull request:
#### What is this?

Another button bellow the editor that allows people to easily insert a module to their article
Once the button get their position inside tinyMCE this will look a lot better.
#### How to test

Apply this PR with patch tester
Go to discover and install the new plugin
Edit an article (fronted/backend) and make sure that the module is inserted
Also insert a style in the given input, either html5 or none, and observe the changes
#### Preview

![screen shot 2015-10-11 at 11 35 32](http://issues.joomla.org/uploads/1/8bb0fda942868e369aa370588c9ca848.png)

dgrammatiko committed:
init

brianteeman commented:
Your text says insert article inside article. ??
On 11 Oct 2015 2:36 pm, "Dimitris Grammatiko" <notifications@github.com>
wrote:

> What is this?
>
> Another button bellow the editor that allows people to easily insert an
> article to their article
> Once the button get their position inside tinyMCE this will look a lot
> better.
> How to test
>
> Apply this PR with patch tester
> Go to discover and install the new plugin
> Edit an article (fronted/backend) and make sure that the module is inserted
> Preview
>
> [image: screenshot 2015-10-11 16 35 09]
> <https://cloud.githubusercontent.com/assets/3889375/10417002/1ffbb48e-7036-11e5-8f46-f2b384587038.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/joomla/joomla-cms/pull/8064
> Commit Summary
>
>    - init
>
> File Changes
>
>    - *A* administrator/components/com_modules/views/modules/tmpl/modal.php
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-0> (206)
>    - *M* administrator/components/com_modules/views/modules/view.html.php
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-1> (5)
>    - *A* administrator/language/en-GB/en-GB.plg_editors-xtd_module.ini
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-2> (8)
>    - *A* administrator/language/en-GB/en-GB.plg_editors-xtd_module.sys.ini
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-3> (10)
>    - *A* components/com_modules/controller.php
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-4> (41)
>    - *A* components/com_modules/modules.php
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-5> (21)
>    - *A* plugins/editors-xtd/module/module.php
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-6> (56)
>    - *A* plugins/editors-xtd/module/module.xml
>    <https://github.com/joomla/joomla-cms/pull/8064/files#diff-7> (19)
>
> Patch Links:
>
>    - https://github.com/joomla/joomla-cms/pull/8064.patch
>    - https://github.com/joomla/joomla-cms/pull/8064.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/8064>.
>

dgt41 commented:
Thanks Brian,

It’s the fever effect I guess<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

dgrammatiko committed:
language should load before sxec the controller

dgrammatiko committed:
CS

brianteeman commented:
If the tinymce update that puts the buttons into the toolbar is accepted
then this is a great addition. If its not then it just adds more clutter in
the UI. my 2c
On 11 Oct 2015 2:39 pm, "Dimitris Grammatiko" <notifications@github.com>
wrote:

> Thanks Brian,
>
> It’s the fever effect I guess
> ------------------------------
> This comment was created with the J!Tracker Application
> <https://github.com/joomla/jissues> at issues.joomla.org/joomla-cms/8064
> <http://issues.joomla.org/tracker/joomla-cms/8064>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/8064#issuecomment-147192718>.
>

dgrammatiko committed:
Only front end modules

dgrammatiko committed:
lang

zero-24 commented on administrator/language/en-GB/en-GB.plg_editors-xtd_module.sys.ini:
```diff
@@ -0,0 +1,10 @@
+PLG_EDITORS-XTD_MODULE="Button - Module"
+PLG_MODULE_XML_DESCRIPTION="Displays a button to make it possible to insert a module into an Article. Displays a popup allowing you to choose the module."
+
+
```
Can we get that clear lines out?

zero-24 commented on plugins/editors-xtd/module/module.php:
```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Editors-xtd.article
```
I guess we should use "module" here?

dgrammatiko committed:
oops

zero-24 commented on plugins/editors-xtd/module/module.php:
```diff
@@ -0,0 +1,56 @@
+	/**
+	 * Display the button
+	 *
+	 * @return array
```
Can we have a @since tag here?

dgrammatiko committed:
CS Thanks @zero-24

zero-24 commented on plugins/editors-xtd/module/module.xml:
```diff
@@ -0,0 +1,19 @@
+	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
+	<authorEmail>admin@joomla.org</authorEmail>
+	<authorUrl>www.joomla.org</authorUrl>
+	<version>1.0.0</version>
```
Version should be 3.5.0

zero-24 commented on plugins/editors-xtd/module/module.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<extension version="3.4" type="plugin" group="editors-xtd" method="upgrade">
```
Same here should be 3.5 ;)

dgrammatiko committed:
improvments

- Insert either with the button on the title of the module or with the position
- View only the published modules
- Field for specyfing the style of the module/position

dgrammatiko committed:
version correction

dgrammatiko committed:
since tag

dgt41 commented:
@brianteeman I totally agree, I wouldn’t create the button if it would be displayed under the editor<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

dgrammatiko committed:
Restrict that input fields

brianteeman commented:
Cool
On 11 Oct 2015 7:28 pm, "Dimitris Grammatiko" <notifications@github.com>
wrote:

> @brianteeman <https://github.com/brianteeman> I totally agree, I wouldn’t
> create the button if it would be displayed under the editor
> ------------------------------
> This comment was created with the J!Tracker Application
> <https://github.com/joomla/jissues> at issues.joomla.org/joomla-cms/8064
> <http://issues.joomla.org/tracker/joomla-cms/8064>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/8064#issuecomment-147233394>.
>

dgrammatiko committed:
exclusions

dgrammatiko committed:
take care of the filters

dgrammatiko committed:
Drag and drop was broken, Fixed!

dgrammatiko committed:
ditch that validation as we are inside an editor

nikosdion commented:
I have tested this item :white_check_mark: successfully on fdec3eabfa23402beb9dc77ce434fa357961d406<br /><br />Works as advertised. I tried both inserting a specific module and a module position. Perfect results. Thank you, Dimitris!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

rdeutz commented:
I have tested this item :white_check_mark: successfully on fdec3eabfa23402beb9dc77ce434fa357961d406<br /><br />works as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

zero-24 commented:
RTC for 3.5. Thanks. :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

infograf768 commented:
@dgt41
This is not multilanguage aware.
Can you filter the available modules in the popup by displaying only those tagged to the current language of the article edited and ALL languages when multilang is enabled? 
A bit as we do for "forcedLanguage" but also adding ALL.

infograf768 commented:
For the frontend, it looks possible by adding
```
		// Modal view should return only specific language and ALL
		if (JFactory::getApplication()->input->get('layout') == 'modal')
		{
			if (JFactory::getApplication()->isSite() && JLanguageMultilang::isEnabled())
			{
				$language = JFactory::getLanguage()->getTag();
				$query->where('a.language in (' . $db->quote(JFactory::getLanguage()->getTag()) . ',' . $db->quote('*') . ')');
			}
		}
```

But for the backend, we would need to check first the language of the article, if already saved.
Any idea?

infograf768 commented:
Corrected code above<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

dgt41 commented:
@infograf768 I guess we can go away with the current status for the backend? Or we provide current state for not saved articles and do the extra filtering in the other occasion

infograf768 commented:
Agree. Can you update PR?

infograf768 commented:
Also, please add the update(s) sql and the new plugin in administrator/language/en-GB/install.xml

zero-24 commented:
Back to pending
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

dgrammatiko committed:
multilingual support. Thanks @infograf768

infograf768 commented:
@dgt41 
Still missing:
>Also, please add the update(s) sql and the new plugin in administrator/language/en-GB/install.xml

zero-24 commented:
As well as the General install sql and the Script.php (com_admin)

dgrammatiko committed:
upd sql files, script

dgrammatiko committed:
lang xml

infograf768 commented:
I have tested this item :white_check_mark: successfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

coolman01 commented:
I have tested this item :red_circle: unsuccessfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<br /><br />I like this - seems far more useful than injecting an article.
Yes it's already possible, but adding a style is cool, and it's definitly simpler to use for newbies.
It worked as decsribed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

coolman01 commented:
I have tested this item :white_check_mark: successfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<br /><br />I like this - seems far more useful than injecting an article.
Yes it's already possible, but adding a style is cool, and it's definitly simpler to use for newbies.
It worked as decsribed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

pascalelocher commented:
I have tested this item :white_check_mark: successfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

Bakual commented:
Set to RTC. Leaving for @roland-d to merge into `3.5-dev` :smile:

roland-d commented:
Thanks everybody, merged into 3.5-dev with commit  06f0d96 

Another hair pulling merge :P
